# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/mnemoc/ktcpvs/make_install.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./Makefile.orig	Sat Feb 22 04:53:17 2003
+++ ./Makefile	Wed Feb 26 14:22:54 2003
@@ -9,17 +9,28 @@
 # uncomment the following line for DEBUG
 DEBUGFLAGS = -g -DCONFIG_TCP_VS_DEBUG
 
+KERNELRELEASE	= $(shell echo -e "\#include <linux/version.h>\nUTS_RELEASE" \
+		    > conftest.c &&					     \
+		    gcc -E -I/usr/src/linux/include conftest.c | tail -1    \
+		    | cut -d '"' -f 2 && rm -f conftest.c)
+
+KERNELSYMS	= /boot/System.map
+
+MODDIR		= kernel/net/ktcpvs/
+MODLIB		= /lib/modules/$(KERNELRELEASE)/$(MODDIR)
 
 CC	= gcc
+DEPMOD	= /sbin/depmod
 CFLAGS	= -D__KERNEL__ -DMODULE  -DEXPORT_SYMTAB -DMODVERSIONS	\
 	$(SMPFLAGS) $(DEBUGFLAGS) -O2 -Wall			\
 	-Wstrict-prototypes -I/usr/src/linux/include		\
 	-include /usr/src/linux/include/linux/modversions.h
 
 LIBS	= regex/librxspencer.a
+TARGETS	= ktcpvs.o ktcpvs_wlc.o ktcpvs_http.o ktcpvs_phttp.o
 
 
-all:	ktcpvs.o ktcpvs_wlc.o ktcpvs_http.o ktcpvs_phttp.o
+all:	$(TARGETS)
 	make -C userspace
 
 ktcpvs.o:	tcp_vs_ctl.o tcp_vs_sched.o tcp_vs_srvconn.o tcp_vs.o	\
@@ -51,6 +62,11 @@
 	insmod ktcpvs.o
 	lsmod
 
+modules_install: $(TARGETS)
+		if [ ! -d "$(MODLIB)" ]; then mkdir -p "$(MODLIB)"; fi
+		install -m 600 -c $(TARGETS) "$(MODLIB)"
+		$(DEPMOD) -ae -F $(KERNELSYMS) $(KERNELRELEASE)
+
 clean:
 	rm -f *.o *~ *.bak *.orig *.rej $(NAME)-$(VERSION).tar.gz
 	make -C userspace clean
